deploy/docker/docker-entrypoint.sh

```diff
@@ -101,103 +101,114 @@ fi
 
 # Fix issue #42 - export env EMQX_DASHBOARD__DEFAULT_USER__PASSWORD to configure
 # 'dashboard.default_user.password' in etc/plugins/emqx_dashboard.conf
-if [[ ! -z "$EMQX_ADMIN_PASSWORD" ]]; then
+if [[ -n "$EMQX_ADMIN_PASSWORD" ]]; then
     export EMQX_DASHBOARD__DEFAULT_USER__PASSWORD=$EMQX_ADMIN_PASSWORD
 fi
 
-# echo value of $VAR_FULL_NAME hiding secrets if any
-echo_value () {
+# echo value of $VAR hiding secrets if any
+echo_value() {
     # get MASK_CONFIG
     MASK_CONFIG_FILTER="$MASK_CONFIG_FILTER|password|passwd|key|token|secret"
-    FORMAT_MASK_CONFIG_FILTER=$(echo $MASK_CONFIG_FILTER |sed -e "s/^[^A-Za-z0-9_]\{1,\}//g"|sed -e "s/[^A-Za-z0-9_]\{1,\}/\|/g")
+    FORMAT_MASK_CONFIG_FILTER=$(echo "$MASK_CONFIG_FILTER" | sed -r -e 's/^[^A-Za-z0-9_]+//' -e 's/[^A-Za-z0-9_]+$//' -e 's/[^A-Za-z0-9_]+/|/g')
 
     # check if contains sensitive value
-    if [ ! -z $(echo $(echo $VAR_NAME | tr '.' ' ') |grep -w -o -E "$FORMAT_MASK_CONFIG_FILTER") ]; then
+    if echo "$VAR_NAME" | grep -iqwE "$FORMAT_MASK_CONFIG_FILTER"; then
         echo "$VAR_NAME=***secret***"
     else
-        echo "$VAR_NAME=$(eval echo \$$VAR_FULL_NAME)"
+        echo "$VAR_NAME=$VAR_VALUE"
     fi
 }
 
 # Catch all EMQX_ prefix environment variable and match it in configure file
-CONFIG="${_EMQX_HOME}/etc/emqx.conf"
-CONFIG_PLUGINS="${_EMQX_HOME}/etc/plugins"
-for VAR in $(env)
-do
+CONFIG="$_EMQX_HOME/etc/emqx.conf"
+CONFIG_PLUGINS="$_EMQX_HOME/etc/plugins"
+for VAR in $(compgen -e); do
     # Config normal keys such like node.name = emqx@127.0.0.1
-    if [[ ! -z "$(echo $VAR | grep -E '^EMQX_')" ]]; then
-        VAR_NAME=$(echo "$VAR" | sed -r "s/EMQX_([^=]*)=.*/\1/g" | tr '[:upper:]' '[:lower:]' | sed -r "s/__/\./g")
-        VAR_FULL_NAME=$(echo "$VAR" | sed -r "s/([^=]*)=.*/\1/g")
-        # Config in emq.conf
-        if [[ ! -z "$(cat $CONFIG |grep -E "^(^|^#*|^#*\s*)$VAR_NAME")" ]]; then
+    if echo "$VAR" | grep -q '^EMQX_'; then
+        VAR_NAME=$(echo "$VAR" | sed -e 's/^EMQX_//' -e 's/__/./g' | tr '[:upper:]' '[:lower:]')
+        VAR_VALUE=${!VAR}
+        # Config in emqx.conf
+        if [[ -n "$(perl -sne 'print if /^[#\s]*\Q$var_name\E\s*=/' -- -var_name="$VAR_NAME" "$CONFIG")" ]]; then
```

Comment:
`grep -E` regard `.` as any char, perl quoted string (`\Q\E`) will escaped the VAR_NAME.